app/controllers/admin/collections_controller.rb

```diff
@@ -30,9 +26,23 @@ def index
       @department =  params[:department]
     end
 
+    scope = scope.order(Arel.sql("#{params[:sort_field]} #{params[:sort_order]}"))
     @collections = scope.page(params[:page]).per(100)
   end
 
+  # Set up click-to-sort column headers.
+  # Note that link_maker is not a component, but
+  # an object that can *return* a component.
+  # This code does NOT perform any actual sorting or filtering.
+  def link_maker
+    @link_maker ||= SortedTableHeaderLinkComponent.link_maker(
+      params: index_params,
+      table_sort_field_key: :sort_field,
+      table_sort_order_key: :sort_order,
+    )
+  end
+  helper_method :link_maker
```

Comment:
This is specifically creating _sort_ links, so let's rename it to make that clear. `sort_link_maker`.  Or with it's possible repsonsibility for adding order to scope too, maybe `sort_link_manager` and class could also be `LinkManager` instead of maker?  Just thoughts.